pkg/server/server_test.go

```diff
@@ -0,0 +1,52 @@
+package server
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+func TestListenAndServe(t *testing.T) {
+	zap.ReplaceGlobals(zap.NewExample())
+
+	tt := []struct {
+		addr   string
+		retErr error
+	}{
+		{
+			addr:   "/tmp/fileNonExists.sock",
+			retErr: nil,
+		},
+		{
+			addr:   "/tmp/fileExists.sock",
+			retErr: nil,
+		},
+		{
+			addr:   ":8080",
+			retErr: nil,
+		},
+	}
+	for _, entry := range tt {
+		ch := make(chan error, 1)
+		s := New()
+		os.Remove(entry.addr)
+		if entry.addr == "/tmp/fileExists.sock" {
+			os.Create(entry.addr)
+			time.Sleep(10 * time.Second)
+		}
+		go func(ch chan error, addr string) {
+			ch <- s.ListenAndServe(addr)
+		}(ch, entry.addr)
+		select {
+		case err := <-ch:
+			if err != entry.retErr {
+				t.Errorf("ListenAndServe() error = %v, wantErr %v", err, entry.retErr)
+			}
+			break
+		case <-time.After(time.Minute):
```

Comment:
Can you reduce this to `time.Second`? a minute is a really long timeout for this test